node/directives/checkAdminAccess.ts

```diff
@@ -47,13 +57,15 @@ export class CheckAdminAccess extends SchemaDirectiveVisitor {
           hasApiToken,
           hasValidApiToken,
           hasStoreToken,
+          hasAdminTokenOnHeader,
+          hasValidAdminTokenOnHeader,
         },
         'CheckAdminAccessAudit'
       )
 
       sendAuthMetric(logger, auditMetric)
 
-      if (!hasAdminToken) {
+      if (!hasAdminToken && !hasAdminTokenOnHeader) {
```

Comment:
just noticed we are missing the !hasApiToken here. Can we add it?
```suggestion
      if (!hasAdminToken && !hasAdminTokenOnHeader && !hasApiToken) {
```

Reply:
Done